File: src_minhaz/wordCount.py
import re
from pyspark import SparkConf, SparkContext


def normalizeWords(text):
    return re.compile(r'\W+', re.UNICODE).split(text.lower())


conf = SparkConf().setMaster("local").setAppName("RatingsHistogram")
sc = SparkContext(conf=conf)

lines = sc.textFile("./Book.txt")
words = lines.flatMap(normalizeWords)
wordCounts = words.map(lambda x: (x, 1)).reduceByKey(lambda x, y: x+y)
wordCountsSorted = wordCounts.map(lambda x: (x[1], x[0])).sortByKey()
results = wordCountsSorted.collect()
for result in results:
    print(result)

File: src/scala/WordCount.scala
import org.apache.spark.SparkContext
import org.apache.spark.SparkConf

object WordCount{
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setMaster("local").setAppName("Word Count")
    val sc = new SparkContext(conf)

    val textFile = sc.textFile("./Book.txt")
    val words = textFile.flatMap(line => line.split(" ")).map(word => (word, 1)).reduceByKey((x, y) => x+y)
    val sortedCountedWords = words.map(x => (x._2, x._1)).sortByKey().map(x => (x._2, x._1))
    sortedCountedWords.foreach(println)
    val results = sortedCountedWords.collect()
    for (result <- results){
      println(result)
    }
  }
}
